cli/azd/internal/scaffold/spec.go

```diff
@@ -98,27 +98,56 @@ type ServiceSpec struct {
 	Name string
 	Port int
 
-	Env map[string]string
+	EnvironmentVariableInformation EnvironmentVariableInformation // todo: merge EnvironmentVariableInformation and env
 
 	// Front-end properties.
 	Frontend *Frontend
 
 	// Back-end properties
 	Backend *Backend
+}
 
-	// Connection to a database
-	DbPostgres    *DatabasePostgres
-	DbMySql       *DatabaseMySql
-	DbRedis       *DatabaseRedis
-	DbCosmosMongo *DatabaseCosmosMongo
-	DbCosmos      *DatabaseCosmosAccount
-
-	// AI model connections
-	AIModels []AIModelReference
+type EnvironmentVariableInformation struct {
+	StringEnvironmentVariables    []StringEnvironmentVariable
+	SecretRefEnvironmentVariables []SecretRefEnvironmentVariable
```

Comment:
we don't need secret ref here in the spec.

Reply:
```go
type secret struct {
  name string
  value string // if the value follows certain pattern, it is a reference
}
```

// secret ref -> { secret key + secret value} // do not define in spec
   secret     -> { name       + value }

```go
func kvSecretName(name string) string {
  return the key of kv secret
}
```

```go
$__value__
```


```go
secret = Secret {
    name: mongo.url
    value: $__mongo.url__,
}
```